src/polycap-photon.c

```diff
@@ -419,6 +430,14 @@ void polycap_photon_free(polycap_photon *photon)
 		free(photon->amu);
 	if (photon->scatf)
 		free(photon->scatf);
+	//TODO: is this correct?
```

Comment:
Geen idee of dit uberhaupt klopt. Ik vermoed van niet gezien ik memory problemen heb, maar weet ook niet hoe ik wel beter aanpak.

Reply:
Is dit stuk dan koosjer? Ik twijfel aan die `if (photon->leaks)` optie. Gaat die enkel triggeren zodra er een `photon->leaks` memory werd geallocceerd, ongeacht hoeveel er in die `leaks` 'array' zitten?

Reply:
zou geen probleem mogen zijn. In `polycap_photon_new` wordt `calloc` gebruikt zodat `photon->leaks` op NULL` wordt gezet. Bij uw eerste leak wordt dat dan gealloceerd zodat het niet meer NULL is en het dus gefree'ed zal worden in `polycap_photon_free`.

Je zou eventueel ook `photon->n_leaks` gebruikt kunnen hebben in de if statement, maar het resultaat zou exact hetzelfde zijn.

Btw uw laatste commit zorgt nog niet voor een tevreden valgrind. Er blijven fouten komen uit `polycap_capil_trace_wall`. Kan je daar eens naar kijken? Momenteel geen tijd... :sad:

Reply:
Zal ernaar kijken. Laat je `valgrind` een test lopen om deze fouten op te sporen, of het volledige programma?

Reply:
Gewoon in de `tests` directory:

`valgrind --track-origins=yes ./source`

De fouten komen onmiddellijk te voorschijn.

Reply:
aja, kwas de `--track-origins=yes` weeral vergeten. I'm on it :)